client/scss/components/_preview-panel.scss

```diff
@@ -0,0 +1,173 @@
+.preview-panel {
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+
+  --preview-width-ratio: min(
+    1,
+    calc(var(--preview-panel-width, 450) / var(--preview-device-width, 375))
+  );
+  --preview-iframe-width: calc(1px * var(--preview-device-width, 375));
+
+  &__area {
+    height: 100%;
+    overflow: hidden;
```

Comment:
This `overflow: hidden` is needed for the warning message when the data is not valid